src/PendingReason_Item.php

```diff
@@ -197,11 +215,51 @@ public function getAutoResolvedate()
             return false;
         }
 
-       // If there was a bump, calculate from last_bump_date
-        $date = new DateTime($this->fields['last_bump_date']);
-        $date->setTimestamp($date->getTimestamp() + $this->fields['followup_frequency'] * ($this->fields['followups_before_resolution'] + 1 - $this->fields['bump_count']));
+        $remaining_bumps = ($this->fields['followups_before_resolution'] > 0 ?
+            $this->fields['followups_before_resolution'] : 0)
+            - $this->fields['bump_count'] + 1;
```

Comment:
I understand that you were forced to use -1 for no followups as 0 is already used to indicate an empty value.
Maybe a comment could help futures readers to understand this constraint.

```suggestion
        // -1 = auto resolution without bumps
        $expected_bumps = min($this->fields['followups_before_resolution'], 0);
        $remaining_bumps = $expected_bumps - $this->fields['bump_count'] + 1;
```